magma/backend/mlir/when_utils.py

```diff
@@ -135,7 +139,11 @@ def __init__(self):
                 self.index = tuple()
 
             def __getitem__(self, idx):
-                self.index += (idx, )
```

Comment:
let's move this class outside of the function so that it doesn't get constructed on every function call. can modify the class name if it could collide with something else.